commands/eval.md

```diff
@@ -194,6 +194,24 @@ returned in the format specified above (as a Lua table with an `err` field).
 The script can pass the exact error to the user by returning the error object
 returned by `redis.pcall()`.
 
+## Running Lua under low memory conditions
+
+When the memory usage in Redis exceeds the `maxmemory` limit, the first write command
+encountered in the Lua script that uses additional memory will cause the script to
+abort. However, one thing to caution here is that if the first write command does
+not use additional memory such as DEL, LREM, or SREM, etc, Redis will allow it to run and
+all subsequent commands in the Lua script will execute to completion in order to preserve
+atomicity of the script execution. If the subsequent writes in the script generate additional
+memory, the Redis memory usage can go over `maxmemory`.
+
+Another possible way for Lua script to cause Redis memory usage to go above `maxmemory`
+happens when the script execution starts when Redis is slightly below `maxmemory` so the
+first write command in the script is allowed. As the script executes, subsequent write
+commands continue to generate memory and causes the Redis server to go above `maxmemory`.
+
+In those scenarios, it is recommended to configure the `maxmemory-policy` appropriately
+to allow the proper evictions of items.
```

Comment:
i didn't understand what you meant by that?
the script will "properly" execute regardless of the maxmemory-policy.
one can even argue that it ignores it and bloats the memory in cases where it shouldn't have.

Reply:
Well, the point I am trying to make here is that in between Lua scripts, Redis can evict items to reclaim memory if the `maxmemory-policy` is configured properly. We have seen some user who has configured `maxmemory-policy` to be `no-eviction` when the Lua script keeps blowing up memory, and there was no way redis can evict items to recover.